### PR TITLE
Extended neva model to support llava_next

### PR DESCRIPTION
**Pull request by yashaswikarnati**
# What does this PR do ?

Extended neva model to support llava_next

**Collection**: 
VLM

# Changelog 
- Utility functions for image tiling
- LLaVA Next model class
- LLaVA Next HF checkpoint importer class
-NeVA fwd function changes to handle image tiling

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [Y] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/vlm/neva/model/base.py [hunk 1]
```diff
@@ -570,18 +575,41 @@ def forward(
...
+        if self.config.is_llava_next:
+            media_embeddings = torch.split(media_embeddings, num_media_tiles.tolist(), dim=0)
+            media_embeddings, feature_lens = pack_image_features(
+                media_embeddings,
+                image_sizes,
+                vision_feature_select_strategy='default',
+                image_newline=self.image_newline,
+            )
+            combined_embeddings, attention_mask, position_ids, final_labels, final_input_ids, final_loss_mask = (
+                merge_input_ids_with_image_features(
```
Comment (yaoyu-33) [inline nemo/collections/vlm/neva/model/base.py:587]: I don't like this design here, reason - 
this method is very similar function as `self._preprocess_data`. In self._preprocess_data we do more things to support sp and cp. It will be good if we can re-use the same method. Can you tell me how these 2 methods are different?

File: nemo/collections/vlm/neva/model/base.py [hunk 2]
```diff
@@ -455,6 +459,7 @@ def forward(
         self,
         input_ids: torch.Tensor,
         position_ids: torch.Tensor,
+        image_sizes: List[torch.Tensor] = None,
```
Comment (yaoyu-33) [inline nemo/collections/vlm/neva/model/base.py:462]: if we are going to change the signiture, it would be better we are using 2 model class? it will might break something we will use nemo / mcore llava

Comment (yashaswikarnati) [reply nemo/collections/vlm/neva/model/base.py:462]: It should not break anything for vanilla neva as image_sizes tensor is optional and will be used only for llava_next. The reason for using the neva model class is most of the functionality is same.
